ethvm/main.cpp

```diff
@@ -144,94 +118,58 @@ int main(int argc, char** argv)
-				networkName = Network::ByzantiumTest;
-			else if (network == "Frontier")
-				networkName = Network::FrontierTest;
-			else if (network == "Ropsten")
-				networkName = Network::Ropsten;
-			else if (network == "Homestead")
-				networkName = Network::HomesteadTest;
-			else if (network == "Main")
-				networkName = Network::MainNetwork;
-			else
-			{
-				cerr << "Unknown network type: " << network << "\n";
-				return -1;
-			}
-		}
-		else if (arg == "stats")
+	po::options_description transactionOptions("Transaction options");
+	transactionOptions.add_options()
+			("value", po::value<string>(), "<n>  Transaction should transfer the <n> wei (default: 0).")
+			("gas", po::value<string>(), "<n>    Transaction should be given <n> gas (default: block gas limit).")
+			("gas-price", po::value<string>(), "<n>  Transaction's gas price' should be <n> (default: 0).")
+			("sender", po::value<string>(), "<a>  Transaction sender should be <a> (default: 0000...0069).")
+			("origin", po::value<string>(), "<a>  Transaction origin should be <a> (default: 0000...0069).")
+			("input", po::value<string>(), "<d>   Transaction code should be <d>")
+			("code", po::value<string>(), "<d>    Contract code <d>. Makes transaction a call to this contract")
+			("gas-limit", po::value<string>(), "");
+	po::options_description vmOptions("VM options");
+	//#if ETH_EVMJIT
+	vmOptions.add_options()
+			("vm", "<vm-kind>  Select VM. Options are: interpreter, jit, smart. (default: interpreter)");
+	//#endif // ETH_EVMJIT
+	po::options_description networkOptions("Network options");
+	networkOptions.add_options()
+			("network",  po::value<string>(), "Main|Ropsten|Homestead|Frontier|Byzantium|Constantinople\n");
+	po::options_description optionsForTrace("Options for trace");
+	optionsForTrace.add_options()
+			("flat", "Minimal whitespace in the JSON.")
+			("mnemonics", "Show instruction mnemonics in the trace (non-standard).\n");
+	po::options_description generalOptions("General options");
+	generalOptions.add_options()
+			("version,v", "Show the version and exit.")
+			("help,h", "Show this help message and exit.")
+			("author", po::value<string>(), "<a> Set author")
+			("difficulty", po::value<string>(), "<d> Set difficulty")
+			("number", po::value<string>(), "<d> Set number")
+			("timestamp", po::value<string>(), "<d> Set timestamp");
+	po::options_description allowedOptions("Usage ethvm <options> [trace|stats|output|test] (<file>|-)");
+	allowedOptions.add(vmOptions).add(networkOptions).add(optionsForTrace).add(generalOptions).add(transactionOptions);
+	po::parsed_options parsed = po::command_line_parser(argc, argv).options(allowedOptions).allow_unregistered().run();
+	vector<string> unrecognisedOptions = collect_unrecognized(parsed.options, po::include_positional);
+	po::variables_map vm;
+	po::store(parsed, vm);
+	po::notify(vm);
+	for (size_t i = 0; i < unrecognisedOptions.size(); ++i) {
+		string arg = unrecognisedOptions[i];
+		if (arg == "stats")
```

Comment:
Why are these handled separately?

Reply:
Because I don't know how to declare options without "--" in the beginning